src/pipelines/imerg_pipeline.py

```diff
@@ -86,12 +86,19 @@ def process_data(self, raw_filename, date):
                 da["time"] = pd.to_datetime(
                     [pd.Timestamp(t.strftime("%Y-%m-%d")) for t in da["time"].values]
                 )
+            date_valid = pd.Timestamp(da['time'].values[0])
             da = da.rename({"lon": "x", "lat": "y"}).squeeze(drop=True)
-            self.metadata["date_valid"] = date.day
-            self.metadata["month_valid"] = date.month
-            self.metadata["year_valid"] = date.year
+            self.metadata["date_valid"] = date_valid.day
+            self.metadata["month_valid"] = date_valid.month
+            self.metadata["year_valid"] = date_valid.year
             da = invert_lat_lon(da)
             da = da.rio.write_crs("EPSG:4326", inplace=False)
+
+            if date_valid != date:
+                filename = self._generate_processed_filename(date_valid)
+                self.logger.warning(f"Date mismatch: date in metadata is {date_valid} and parameter date is {date}."
```

Comment:
This should also probably be more than a warning. The pipeline should fail in this case.

Reply:
And it feels a bit convoluted to call `self._generate_processed_filename()` potentially twice. What about just dropping [this line](https://github.com/OCHA-DAP/ds-raster-pipelines/blob/ba68cd0e37c5c55882f0d639b74feb828fa21fd1/src/pipelines/imerg_pipeline.py#L76) and adjusting this block slightly to: 

```
filename = self._generate_processed_filename(date_valid)
if date_valid != date:
  self.logger.error(f"Date mismatch: date in metadata is {date_valid} and parameter date is {date}.")
```